src/Land/Model/source.jl

```diff
@@ -13,6 +39,51 @@ function land_source!(
     f(land, source, state, diffusive, aux, t, direction)
 end
 
+function land_source!(
+    source_type::PhaseChange,
+    land::LandModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    FT = eltype(state)
+
+    _ρliq = FT(ρ_cloud_liq(land.param_set))
+    _ρice = FT(ρ_cloud_ice(land.param_set))
+    _Tfreeze = FT(T_freeze(land.param_set))
+    _LH_f0 = FT(LH_f0(land.param_set))
+    _g = FT(grav(land.param_set))
+
+    ϑ_l, θ_i = get_water_content(land.soil.water, aux, state, t)
+    eff_porosity = land.soil.param_functions.porosity - θ_i
+    θ_l = volumetric_liquid_fraction(ϑ_l, eff_porosity)
+    T = get_temperature(land.soil.heat, aux, t)
+
+    # Take absolute value so that inverse_matric_potential can return a value
+    # but below, use heaviside function to only allow freezing in the
+    # appropriate temperature range
+    ψ = -abs(_LH_f0 / _g / _Tfreeze * (T - _Tfreeze))
+    hydraulics = land.soil.water.hydraulics
+    θstar =
+        land.soil.param_functions.porosity *
+        inverse_matric_potential(hydraulics, ψ)
+
+    τft = max(source_type.Δt, source_type.τLTE)
+    freeze_thaw =
+        1.0 / τft * (
```

Comment:
```suggestion
        1 / τft * (
```
(or `FT(1)`, but `1/A*B` will promote properly).